### DISCUSSION TITLE
WGPU?

### DISCUSSION THREAD

**Discussion by C-BJ** [inanevin/LinaEngine#67] [Ideas]
Continue https://github.com/inanevin/LinaEngine/issues/56's topic
I think you may know that, just to clarify, wgpu is low-level as Vulkan, metal, DX, etc

**Comment by C-BJ** [comment]
With wgpu, you can use wgsl as modern shader language at the same time

**Comment by inanevin** [comment]
The reason for not using WGPU is the same as not using BGFX, Google Filament etc. This will create a dependency to an external tool/project. These dependencies are fine as long as they are created only on a small part of the code-base. However a large part, such as Rendering, is way too much work to refactor/rewrite when there's a problem on the dependent external code.

Having your own rendering abstraction/framework also provides me the ability to customize/improve/enhance as much as I want as well as fix bugs a lot more quickly.
